src/test/java/edu/hm/hafner/util/SetTest.java

```diff
@@ -0,0 +1,198 @@
+    }
+
+    @Test
+    void shouldReturnIterator() {
+        HashSet<Integer> testSet = creatHashSet(3);
+        Iterator<Integer> testIterator = testSet.iterator();
+        HashSet<Integer> emptyTestSet = creatHashSet(0);
+        Iterator<Integer> emptyTestIterator = emptyTestSet.iterator();
+        HashSet<Integer> testSetWithNull = createHashSetWithOnlyNull();
+        Iterator<Integer> nullIterator = testSetWithNull.iterator();
+
+        Assertions.assertThat(testIterator.hasNext()).isTrue();
+        Assertions.assertThat(testIterator.next()).isBetween(0,2);
+        Assertions.assertThat(testIterator.hasNext()).isTrue();
+        Assertions.assertThat(testIterator.next()).isBetween(0,2);
+        Assertions.assertThat(testIterator.hasNext()).isTrue();
+        Assertions.assertThat(testIterator.next()).isBetween(0,2);
+        Assertions.assertThat(testIterator.hasNext()).isFalse();
+
+        Assertions.assertThat(emptyTestIterator.hasNext()).isFalse();
+
+        Assertions.assertThat(nullIterator.hasNext()).isTrue();
+    }
+
+    @Test
+    void shouldReturnCorrectSize() {
+        HashSet<Integer> testSetZero = creatHashSet(0);
+        HashSet<Integer> testSetOne = creatHashSet(1);
+        HashSet<Integer> testSetTwo = creatHashSet(2);
+        HashSet<Integer> testSetHundred = creatHashSet(100);
+        HashSet<Integer> testSetWithNull = createHashSetWithNullAndSize4();
+
+        Assertions.assertThat(testSetZero.size() == 0).isTrue();
+        Assertions.assertThat(testSetOne.size() == 1).isTrue();
+        Assertions.assertThat(testSetTwo.size() == 2).isTrue();
+        Assertions.assertThat(testSetHundred.size() == 100).isTrue();
+        Assertions.assertThat(testSetWithNull.size() == 4).isTrue();
+    }
+
+    @Test
+    void shouldTestForEmptiness() {
+        HashSet<Integer> testSetEmpty = creatHashSet(0);
+        HashSet<Integer> testSetSizeOne = creatHashSet(4);
+        HashSet<Integer> testSetWithNull = createHashSetWithOnlyNull();
+
+        Assertions.assertThat(testSetEmpty.isEmpty()).isTrue();
+        Assertions.assertThat(testSetSizeOne.isEmpty()).isFalse();
+        Assertions.assertThat(testSetWithNull.isEmpty()).isFalse();
+    }
+
+    @Test
+    void shouldTestForContainment() {
+        HashSet<Integer> testSet = createHashSetWithNullAndSize4();
+
+        Assertions.assertThat(testSet.contains(0)).isTrue();
+        Assertions.assertThat(testSet.contains(1)).isTrue();
+        Assertions.assertThat(testSet.contains(2)).isTrue();
+        Assertions.assertThat(testSet.contains(null)).isTrue();
+        Assertions.assertThat(testSet.contains(3)).isFalse();
+        Assertions.assertThat(testSet.contains(-1)).isFalse();
+    }
+
+    @Test
+    void shouldAddItem() {
+        HashSet<Integer> testSet = creatHashSet(0);
+        HashSet<Integer> testSetWithNull = createHashSetWithNullAndSize4();
+
+        // add string, add existing item
+        Assertions.assertThat(testSet.isEmpty()).isTrue();
```

Comment:
Immer die AssertJ assertions nutzen, statt `isTrue`:
```suggestion
        Assertions.assertThat(testSet).isEmpty();
```